pkg/pillar/cmd/zedagent/handlecipherconfig.go

```diff
@@ -115,10 +115,10 @@ func parseCipherBlock(ctx *getconfigContext, key string, cfgCipherBlock *zcommon
 
 	// get CipherContext and embed it into CipherBlockStatus to avoid potential races
 	for _, cfgCipherContext := range ctx.cipherContexts {
-		if cfgCipherContext.ContextID != cipherBlock.CipherContextID {
-			continue
+		if cfgCipherContext.ContextID == cipherBlock.CipherContextID {
+			cipherBlock.CipherContext = &cfgCipherContext
+			break
```

Comment:
For these cases, I prefer using the index in the loop, breaking on a match and then getting the object by the index. For me, it looks more explicit. But that's also fine.

Reply:
I would prefer to reference the copy here instead of the actual array element, in case the array might be subject to change.